### PR TITLE
Console: do not store enabled status for unknown thing

### PR DESCRIPTION
**Pull request by htreu**
Follow up on #6604.

Signed-off-by: Henning Treu <henning.treu@googlemail.com>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/console/ThingConsoleCommandExtension.java [hunk 1]
```diff
@@ -141,6 +141,10 @@ private void removeAllThings(Console console, Collection<Thing> things) {
     }
 
     private void enableThing(Console console, ThingUID thingUID, boolean isEnabled) {
+        if (thingRegistry.get(thingUID) == null) {
+            console.println("unknwon thing for thingUID '" + thingUID.getAsString() + "'.");
```
Comment (itn3rd77) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/console/ThingConsoleCommandExtension.java]: Typo: unknwon should be unknown

Comment (htreu) [reply bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/console/ThingConsoleCommandExtension.java]: tnx.

**Comment by maggu2810** [comment]
Just another idea: Why not changing the return value of the `setEnabled` method to a boolean if the value has been changed? As we dropped the IAE already a boolean instead of a void return type should not matter (does it?)

**Comment by htreu** [comment]
@maggu2810 not sure if I understand you correct. This PR is only about the console command to not allow persisting enablement information for arbitrary thingUIDs.
Are you talking about the ThingManager#setEnabled?

**Comment by maggu2810** [comment]
You try to get the thing with a given UID from the registry just to check if the result is null -- so to check if there is such a thing. If not, you print an information and return.
If not the `setEnabled` does the same again.
It tries to get the thing for the UID and return is there is no such thing.
There is no need to do the same logic twice as `setEnabled` could return with the indication that it cannot proceed. And the thing could be removed between the check and the call to `setEnabled`.

Both does not "really" harm
* the additional overhead does not matter for a console command
* the potential thing removal between that calls can be neglected

But if we can make it "better" why not adding a return value and use this one to generate that method instead of adding additional code that does the same again (before).

**Comment by htreu** [comment]
> the setEnabled does the same again.

Since #6604 this is nit the case anymore. From the API we accept now any thingUID, even if the ThingManager is not aware of the thing itself.
This PR prevents the console user from adding arbitrary thingUID information to the enablement database.

**Comment by maggu2810** [comment]
You are right, the `getThing` method that is used in `setEnabled` of the thing manager implementation does not lookup using the registry but the self maintained thing collection that is controlled by the events (thing added, ...).
But shouldn't this be equal to the registry with respect to some timing...

**Comment by htreu** [comment]
> But shouldn't this be equal to the registry with respect to some timing...

#6604 is all about timing. If you follow the discussion there it is about order of calling ThingRegistryListeners and ThingTrackers (the ThingManager).

Thanks for merging this :-)

**Comment by maggu2810** [comment]
I followed it already -- but perhaps I missed something or I just don't really understand this PR.

Without this PR it it possible to enable / disable a thing using the console extension if the thing UID is not known by the thing registry but known by the thing manager implementation maintained thing collection.

This is how the thing manager implementation works (now). It does not rely on the thing registry knowledge but maintains its own information base.

So, why does it make sense to add that check on this entry point?
Do we need to add that check to every possible entry point (REST API, other programmatic usage in products, ...) or shouldn't the caller rely that the thing manager implementation does handle the set enabled itself so we don't need to do some pre-checks.

**Comment by htreu** [comment]
> Without this PR it it possible to enable / disable a thing using the console extension if the thing UID is not known by the thing registry but known by the thing manager implementation maintained thing collection.

No, w/o this PR it would be possible to tell the ThingManager to persists enablement information for any thingUID.
See:

https://github.com/eclipse/smarthome/blob/20ac7cfede61f8d367a715d3eb745f52416233e5/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java#L1209-L1214

It will persist the information w/o checking if the thing actually exists.

**Pull request merged**